packages/yuudachi/src/events/manual-report-status/threadUpdate.ts

```diff
@@ -0,0 +1,70 @@
+import { on } from "node:events";
+import type { ThreadChannel } from "discord.js";
+import { Client, Events } from "discord.js";
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+import type { Sql } from "postgres";
+import { inject, injectable } from "tsyringe";
+import type { Event } from "../../Event.js";
+import { resolveLabelToStatus, resolveStatusLabel, upsertReportLog } from "../../functions/logging/upsertReportLog.js";
+import type { RawReport } from "../../functions/reports/transformReport.js";
+import { transformReport } from "../../functions/reports/transformReport.js";
+import { updateReport } from "../../functions/reports/updateReport.js";
+import type { ReportLabels } from "../../functions/settings/getGuildSetting.js";
+import { REPORT_STATUS_KEYS, getGuildSetting, SettingsKeys } from "../../functions/settings/getGuildSetting.js";
+import { logger } from "../../logger.js";
+import { kSQL } from "../../tokens.js";
+
+@injectable()
+export default class implements Event {
+	public name = "Manual report status change";
+
+	public event = Events.ThreadUpdate as const;
+
+	public constructor(public readonly client: Client<true>, @inject(kSQL) public readonly sql: Sql<any>) {}
+
+	public async execute(): Promise<void> {
+		for await (const [oldPost, newPost] of on(this.client, this.event) as AsyncIterableIterator<
+			[ThreadChannel, ThreadChannel]
+		>) {
+			try {
+				const reportLabels = await getGuildSetting<ReportLabels>(oldPost.guildId, SettingsKeys.ReportLabels);
```

Comment:
Shouldn't this check the parent channel to prevent bugs with normal threads and other forums?

Reply:
don't see at which point that could fail, as those updates will never have report tags

Reply:
That's true 🤔